tools/bridge/bridge/event_fetcher.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import gevent
+
+from typing import List
+from gevent.queue import Queue
+from web3 import Web3
+
+
+class EventFetcher:
+    def __init__(
+        self,
+        *,
+        web3: Web3,
+        contract_address: str,
+        event_signature: str,
+        event_argument_filter: List,
+        event_fetch_limit: int = 950,
+        event_queue: Queue,
```

Comment:
Because of the lack for a fitting type and a custom abstract class would a overshoot, I left it untyped for now. Thereby it does not restrict the interface like it does now.